Mathlib/Probability/Kernel/Disintegration/MeasurableStieltjes.lean

```diff
@@ -0,0 +1,494 @@
+section DefaultRatCDF
+
+/-- A function with the property `IsRatCDF`.
+Used in a piecewise construction to convert a function which only satisfies the properties
+defining `IsRatCDF` on some set into a true `IsRatCDF`. -/
+def defaultRatCDF (q : ℚ) := if q < 0 then (0 : ℝ) else 1
+
+lemma monotone_defaultRatCDF : Monotone defaultRatCDF := by
+  unfold defaultRatCDF
+  intro x y hxy
+  dsimp only
+  split_ifs with h_1 h_2 h_2
+  exacts [le_rfl, zero_le_one, absurd (hxy.trans_lt h_2) h_1, le_rfl]
+
+lemma defaultRatCDF_nonneg (q : ℚ) : 0 ≤ defaultRatCDF q := by
+  unfold defaultRatCDF
+  split_ifs
+  exacts [le_rfl, zero_le_one]
+
+lemma defaultRatCDF_le_one (q : ℚ) : defaultRatCDF q ≤ 1 := by
+  unfold defaultRatCDF
+  split_ifs <;> simp
+
+lemma tendsto_defaultRatCDF_atTop : Tendsto defaultRatCDF atTop (𝓝 1) := by
+  refine (tendsto_congr' ?_).mp tendsto_const_nhds
+  rw [EventuallyEq, eventually_atTop]
+  exact ⟨0, fun q hq => (if_neg (not_lt.mpr hq)).symm⟩
+
+lemma tendsto_defaultRatCDF_atBot : Tendsto defaultRatCDF atBot (𝓝 0) := by
+  refine (tendsto_congr' ?_).mp tendsto_const_nhds
+  rw [EventuallyEq, eventually_atBot]
+  refine ⟨-1, fun q hq => (if_pos (hq.trans_lt ?_)).symm⟩
+  linarith
+
+lemma iInf_rat_gt_defaultRatCDF (t : ℚ) :
+    ⨅ r : Ioi t, defaultRatCDF r = defaultRatCDF t := by
+  simp only [defaultRatCDF]
+  have h_bdd : BddBelow (range fun r : ↥(Ioi t) ↦ ite ((r : ℚ) < 0) (0 : ℝ) 1) := by
+    refine ⟨0, fun x hx ↦ ?_⟩
+    obtain ⟨y, rfl⟩ := mem_range.mpr hx
+    dsimp only
+    split_ifs
+    exacts [le_rfl, zero_le_one]
+  split_ifs with h
+  · refine le_antisymm ?_ (le_ciInf fun x ↦ ?_)
+    · obtain ⟨q, htq, hq_neg⟩ : ∃ q, t < q ∧ q < 0 := ⟨t / 2, by linarith, by linarith⟩
+      refine (ciInf_le h_bdd ⟨q, htq⟩).trans ?_
+      rw [if_pos]
+      rwa [Subtype.coe_mk]
+    · split_ifs
+      exacts [le_rfl, zero_le_one]
+  · refine le_antisymm ?_ ?_
+    · refine (ciInf_le h_bdd ⟨t + 1, lt_add_one t⟩).trans ?_
+      split_ifs
+      exacts [zero_le_one, le_rfl]
+    · refine le_ciInf fun x ↦ ?_
+      rw [if_neg]
+      rw [not_lt] at h ⊢
+      exact h.trans (mem_Ioi.mp x.prop).le
+
+lemma isRatStieltjesPoint_defaultRatCDF (a : α) :
+    IsRatStieltjesPoint (fun (_ : α) ↦ defaultRatCDF) a where
+  mono := monotone_defaultRatCDF
+  tendsto_atTop_one := tendsto_defaultRatCDF_atTop
+  tendsto_atBot_zero := tendsto_defaultRatCDF_atBot
+  iInf_rat_gt_eq := iInf_rat_gt_defaultRatCDF
+
+lemma IsRatCDF_defaultRatCDF (α : Type*) [MeasurableSpace α] :
+    IsRatCDF (fun (_ : α) (q : ℚ) ↦ defaultRatCDF q) where
+  isRatStieltjesPoint := isRatStieltjesPoint_defaultRatCDF
+  measurable := measurable_const
+
+end DefaultRatCDF
+
+section ToRatCDF
+
+variable {f : α → ℚ → ℝ}
+
+open Classical in
+/-- Turn a function `f : α → ℚ → ℝ` into another with the property `IsRatCDF`.
+Mainly useful when `f` satisfies the property `IsRatStieltjesPoint f a` almost everywhere. -/
+noncomputable
+def toRatCDF (f : α → ℚ → ℝ) : α → ℚ → ℝ := fun a ↦
```

Comment:
With the current naming, this name is a little bit misleading as the result does not satisfy `IsRatCF` (it is not measurable). But if you change to `IsMeasRatCDF`, and keep `toRatCDF` for this one, then everything is fine :-)